### ISSUE TITLE
Transform rows in topological order (DSL)

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#691]
## Problem

Below, the `regression1_*` targets should not carry the `regression2` trace. Current behavior:

``` r
library(drake)
plan <- drake_plan(
  report = knit(knitr_in("report.Rmd"), file_out("report.md"), quiet = TRUE),
  small = simulate(48),
  large = simulate(64),
  regression1 = target(
    reg1(data),
    transform = map(data = c(small, large), .tag_out = reg),
  ),
  regression2 = target(
    reg2(data),
    transform = map(data, .tag_out = reg),
  ),
  trace = TRUE
)regression1_small

plan[, c("target", "regression1", "regression2")]
#> # A tibble: 7 x 3
#>   target            regression1       regression2      
#>   <chr>             <chr>             <chr>            
#> 1 report            <NA>              <NA>             
#> 2 small             <NA>              <NA>             
#> 3 large             <NA>              <NA>             
#> 4 regression1_small regression1_small <NA>             
#> 5 regression1_large regression1_large <NA>             
#> 6 regression2_large regression1_large regression2_large
#> 7 regression2_small regression1_small regression2_small
```

<sup>Created on 2019-01-24 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Desired result:

```r
#> # A tibble: 7 x 3
#>   target            regression1       regression2      
#>   <chr>             <chr>             <chr>            
#> 1 report            <NA>              <NA>             
#> 2 small             <NA>              <NA>             
#> 3 large             <NA>              <NA>             
#> 4 regression1_small regression1_small <NA>             
#> 5 regression1_large regression1_large <NA>             
#> 6 regression2_large <NA>              regression2_large
#> 7 regression2_small <NA>              regression2_small
```

## Likely solution

Recursive strategy: each stage within `transform_plan()` should independently transform everything with old grouping variables all already in the plan. This should be fast enough to not need parallelization, so there should be no need for an explicit dependency graph or scheduling.

**Comment by wlandau** [comment]
So in that example, I should have written `map(data = c(small, large), ...)` in both cases. `map(data)` already assumes the `data` column is reused. But there are other examples where we need `drake` to process transformations in topological order.

**Comment by wlandau** [comment]
Fixed in #692.
